Java/sort/MergeSortRecursion.java

```diff
@@ -0,0 +1,107 @@
+//merge sort using divide and conquer and recursion
+import java.util.Scanner;
+class MergeSortRecursion 
+{
+    
+    public static void sort(int arr[], int start, int end)
+    {
+        if(start==end) //base case
+        return;
+        else
+             {
+            // Find the mid of the array
+            int mid = (start + end) / 2;
```

Comment:
@ErzaTitania-2001 Please avoid this technique to find the mid element since this can cause an integer overflow. Use this int `int mid = start + ((end - start) / 2);`.